harness/determined/keras/_tf_keras_trial.py

```diff
@@ -72,6 +72,32 @@ def load_optimizer_weights(model: Model, load_path: pathlib.Path) -> None:
             )
 
 
+class DeterminedEarlyStoppingCallback(tf.keras.callbacks.Callback):  # type: ignore
+    """
+    DeterminedEarlyStoppingCallback converts a stop request, so that Determined
+    can handle the stop request by finishing the step and checkpointing.
+    """
+
+    def __init__(self, tf_keras_trial_controller: "TFKerasTrialController") -> None:
+        self.tf_keras_trial_controller = tf_keras_trial_controller
+
+    def _convert_stop_training(self) -> None:
+        # We use stop_training to exit out of the training loop, but we set
+        # expect_terminate when we do so.
+        if (
+            self.tf_keras_trial_controller.model.stop_training
```

Comment:
non-blocking: can you use `self.model` instead of `self.tf_keras_trial_controller.model`. Should be pointing to the same object.